python/lsst/verify/gen2tasks/metricTask.py

```diff
@@ -21,257 +21,44 @@
-        -----
-        The default implementation extracts a
-        `~lsst.pipe.base.PipelineTaskConnections` object from ``config``.
-        """
-        # Get connections from config for backward-compatibility
-        connections = config.connections.ConnectionsClass(config=config)
-        return {name: getattr(connections, name).name
-                for name in connections.inputs}
-
-    @classmethod
-    def areInputDatasetsScalar(cls, config):
-        """Return input dataset multiplicity.
-
-        Parameters
-        ----------
-        config : ``cls.ConfigClass``
-            Configuration for this task.
-
-        Returns
-        -------
-        datasets : `Dict` [`str`, `bool`]
-            Dictionary where the key is the name of the input dataset (must
-            match a parameter to `run`) and the value is `True` if `run` takes
-            only one object and `False` if it takes a list.
-
-        Notes
-        -----
-        The default implementation extracts a
-        `~lsst.pipe.base.PipelineTaskConnections` object from ``config``.
-        """
-        connections = config.connections.ConnectionsClass(config=config)
-        return {name: not getattr(connections, name).multiple
-                for name in connections.inputs}
-
-    @classmethod
-    @abc.abstractmethod
-    def getOutputMetricName(cls, config):
-        """Identify the metric calculated by this ``MetricTask``.
-
-        Parameters
-        ----------
-        config : ``cls.ConfigClass``
-            Configuration for this ``MetricTask``.
-
-        Returns
-        -------
-        metric : `lsst.verify.Name`
-            The name of the metric computed by objects of this class when
-            configured with ``config``.
-        """
-
-    def addStandardMetadata(self, measurement, outputDataId):
-        """Add data ID-specific metadata required for all metrics.
+import abc
 
-        This method currently does not add any metadata, but may do so
-        in the future.
+from deprecated.sphinx import deprecated
 
-        Parameters
-        ----------
-        measurement : `lsst.verify.Measurement`
-            The `~lsst.verify.Measurement` that the metadata are added to.
-        outputDataId : ``dataId``
-            The data ID to which the measurement applies, at the appropriate
-            level of granularity.
+from ..tasks.metricTask import MetricTask as NewTask
 
-        Notes
-        -----
-        This method should not be overridden by subclasses.
 
-        This method is not responsible for shared metadata like the execution
-        environment (which should be added by this ``MetricTask``'s caller),
-        nor for metadata specific to a particular metric (which should be
-        added when the metric is calculated).
+# Docstring needs to have no leading whitespace for @deprecated to work
+@deprecated(
+    reason="Replaced by `lsst.verify.tasks.MetricTask`. "
+           "To be removed along with Gen 2 support.",
```

Comment:
Does this mean "when daf_persistence is removed" or does it mean "when this package is entirely migrated to gen 3"?

Reply:
The former; will clarify.